ar71xx: add VLAN switch to target ENS202EXT

mcprat opened the pull request:
The target ENS202EXT was just recently added right before the stable release of Openwrt 18.

It flashes fine, but the physical switch is almost impossible to use until you have a VLAN set up. Tested on two devices.

The actual problem is that eth0 represents nothing for whatever reason. In other words, both WAN and LAN are running from eth1. There may be an underlying problem in the build, but for now, I assume that this is correct and that a VLAN switch is an appropriate fix.

Also, it's virtually impossible to get the switch running right through LuCI. It is one thing to get a switch to appear, but attempting to configure it breaks the whole thing. The VLAN has to be set up perfectly, otherwise, interfaces will not start up, and one is forced to reset settings, OR, the new LuCI feature kicks in and reverses any steps. It is extremely difficult to determine which virtual ports correspond to which physical ethernet ports without being able to set up the switch in LuCI.

Temporary Workaround:
followed directions here
https://github.com/openwrt/luci/issues/867

Signed-off-by: Michael Pratt <mpratt51@gmail.com>

Commit:
ar71xx: add VLAN switch to target ENS202EXT

The target ENS202EXT was just recently added right before the stable release of Openwrt 18.

It flashes fine, but the physical switch is almost impossible to use until you have a VLAN set up. Tested on two devices.

Temporary Workaround:
followed directions here
https://github.com/openwrt/luci/issues/867

ynezz commented:
@mpratt14 Your PR/commit has formal issues, like missing your `Signed-off-by`, please read https://openwrt.org/submitting-patches

Other then that, your commit message is very sparse, please add the details from this PR's description  to the commit message, so it's clear what the problem is, why is this change needed.

ynezz commented on target/linux/ar71xx/base-files/etc/board.d/02_network:
```diff
@@ -191,7 +191,11 @@ ar71xx_setup_interfaces()
 	dlan-hotspot|\
 	dlan-pro-500-wp|\
-	ens202ext|\
+	ens202ext)
```
@mpratt14 This approach for fixing this issue for ens202ext is wrong, as you're effectively breaking switch configuration for other devices.

mpratt14 replied:
Can you please explain? By "other devices" do you mean other boards, and how would my commit affect any other board in the list?

If you're talking about existing ENS202EXT that other people have, I still can't see the consequence of this change. Even if this would be applied because of a sysupgrade image and nvram/config is to be kept, it would be a step someone else would otherwise have to do manually. Ethernet was nonexistent until I did exactly this and then regenerate board.json. Is that not the reason that there are other customized entries for other boards? Is that not the reason this file exists in the build?

If I'm mistaken please fully inform me,
Thanks

ynezz replied:
@mpratt14 Situation before your patch was, that all0258n, all0315n, dlan-hotspot, dlan-pro-500-wp, ens202ext, ja76pf2, rocket-m-ti, ubnt-unifi-outdoor devices would all use `ucidef_set_interface_lan "eth0 eth1"`

Situation after your patch is, that all0258n, all0315n, dlan-hotspot, dlan-pro-500-wp and ens202ext would use your new switch/vlan config, and ja76pf2, rocket-m-ti, ubnt-unifi-outdoor would use the `ucidef_set_interface_lan "eth0 eth1"`.

In other words, your patch as it is, is probably going to break network setup on all0258n, all0315n, dlan-hotspot and dlan-pro-500-wp devices.

Is that more clear now?

mpratt14 replied:
Sorry, I'm pretty new with all this. Thanks for expanding but that isn't really an explanation of why this is true. After looking very carefully I recognized the or operator "|" and I see now that this is a list of very large if-then statements, and the target ENS202EXT was already part of a statement with other routers in it, and that my commit made the statement end early with different information....

About to pass another commit to put the target at the end before the wildcard.

Thanks

mcprat reviewed (commented):
I can't edit the commit message from the website

ynezz commented:
> I can't edit the commit message from the website

https://openwrt.org/submitting-patches#working_with_github

mpratt14 commented:
> > I can't edit the commit message from the website
> 
> @mpratt14 https://openwrt.org/submitting-patches#working_with_github

I would but I dont have access to a linux machine right now

Commit:
ar71xx: add VLAN switch to target ENS202EXT

(Correcting previous commit that attempted to do this but was not in proper code format)

The target ENS202EXT was just recently added right before the stable release of Openwrt 18.

It flashes fine, but the physical switch is almost impossible to use until you have a VLAN set up. Tested on two devices.

The actual problem is that eth0 represents nothing for whatever reason. In other words, both WAN and LAN are running from eth1. There may be an underlying problem in the build, but for now, I assume that this is correct and that a VLAN switch is an appropriate fix.

Also, it's virtually impossible to get the switch running right through LuCI. It is one thing to get a switch to appear, but attempting to configure it breaks the whole thing. The VLAN has to be set up perfectly, otherwise, interfaces will not start up, and one is forced to reset settings, OR, the new LuCI feature kicks in and reverses any steps. It is extremely difficult to determine which virtual ports correspond to which physical ethernet ports without being able to set up the switch in LuCI.

Temporary Workaround:
followed directions here
[openwrt/luci#867](https://github.com/openwrt/luci/issues/867)

Signed-off-by: Michael Pratt [mpratt51@gmail.com](mailto:mpratt51@gmail.com)

ynezz commented:
> I would but I dont have access to a linux machine right now

Git is usable on almost every operating system today.
